app/frontend/src/components/TOS.tsx

```diff
@@ -1,35 +1,39 @@
-import { Link } from "@material-ui/core";
-import React from "react";
+import { CircularProgress } from "@material-ui/core";
+import Markdown from "components/Markdown";
+import { Error as GrpcError } from "grpc-web";
+import { GetTermsOfServiceRes } from "pb/resources_pb";
+import { tosQueryKey } from "queryKeys";
+import { useQuery } from "react-query";
+import { service } from "service";
+import makeStyles from "utils/makeStyles";
 
-import PageTitle from "./PageTitle";
-import TextBody from "./TextBody";
+const useStyles = makeStyles((theme) => ({
+  root: {
+    maxWidth: theme.breakpoints.values.lg,
+    margin: "0 auto",
+    padding: theme.spacing(2),
+  },
+}));
 
 export default function TOS() {
-  return (
-    <>
-      <PageTitle>Terms of Service</PageTitle>
-      <TextBody>
-        This is a test TOS. I agree to follow the&nbsp;
-        <Link href="https://community.couchers.org/faq">
-          Couchers guidelines
-        </Link>
-        . I understand this is a preview of couchers.org and my data may be
-        erased. Lorem Ipsum is simply dummy text of the printing and typesetting
-        industry.
-        <br />
-        <br />
-        Lorem Ipsum has been the industry's standard dummy text ever since the
-        1500s, when an unknown printer took a galley of type and scrambled it to
-        make a type specimen book. It has survived not only five centuries, but
-        also the leap into electronic typesetting, remaining essentially
-        unchanged.
-        <br />
-        <br />
-        It was popularised in the 1960s with the release of Letraset sheets
-        containing Lorem Ipsum passages, and more recently with desktop
-        publishing software like Aldus PageMaker including versions of Lorem
-        Ipsum.
-      </TextBody>
-    </>
-  );
+  const classes = useStyles();
+  const { data, error, isLoading } = useQuery<
+    GetTermsOfServiceRes.AsObject,
+    GrpcError
+  >({
+    queryKey: tosQueryKey,
+    queryFn: () => service.resources.getTermsOfService(),
+  });
+
+  if (error) {
+    // Re-throw error to trigger error boundary to encourage user to report it
+    // if they can't see the terms
+    throw error;
+  }
+
+  return isLoading ? (
+    <CircularProgress />
+  ) : data ? (
+    <Markdown className={classes.root} source={data?.termsOfService} />
```

Comment:
I think this needs a `<PageTitle>Terms of Service</PageTitle>`, sorry for missing this earlier